### PR TITLE
feat: add debugger integration

### PR DESCRIPTION
**Pull request by dariusbakunas**
Terraform supports using debuggers like `delve` to debug providers, this is following their examples on how to integrate it:

https://www.terraform.io/docs/extend/debugging.html

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

This does not affect any resources or data sources, it only adds `debug` flag support in main function

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -9,8 +11,24 @@ import (
...
+	flag.BoolVar(&debugMode, "debug", false, "set to true to run the IBM terraform provider with support for debuggers like delve")
+	flag.Parse()
+
+	opts := &plugin.ServeOpts{
 		ProviderFunc: ibm.Provider,
-	})
+	}
+
+	if debugMode {
+		err := plugin.Debug(context.Background(), "IBM-Cloud/ibm", opts)
```
Comment (kavya498) [inline main.go:26]: should it not be `registry.terraform.io/IBM-Cloud/ibm` instead of `IBM-Cloud/ibm`?

File: GNUmakefile [hunk 2]
```diff
@@ -15,6 +15,9 @@ tools:
 build: fmtcheck vet
 	go install
 
+build-dbg:
```
Comment (kavya498) [inline GNUmakefile:18]: why is this required in MAKEfile?

File: main.go (hunk 1 above)
Comment (dariusbakunas) [reply main.go:26]: I tested with `IBM-Cloud/ibm` but I can test with full url too

File: GNUmakefile (hunk 2 above)
Comment (dariusbakunas) [reply GNUmakefile:18]: you need to disable compiler optimizations to produce debuggable binary, so you need to pass thos `gcflags`. It is easier to remember `make build-dbg` vs `go install -gcflags "all=-N -l"`

File: main.go (hunk 1 above)
Comment (dariusbakunas) [reply main.go]: Looks like it works both ways, so I changed to full url

**Comment by dariusbakunas** [comment]
Maybe I should've opened this as feature request first? :) I don't mind closing it if you don't think this would be useful at this point, but I find it is sometimes easier to look at values during runtime instead of trying to find issue with logs, either way, open for suggestions
